### ISSUE TITLE
Session.refer with replaces terminates immediately on a 2** response, no matter what the actual REFER result was. 

### ISSUE THREAD

**Issue by klarrimore** [onsip/SIP.js#254]
Session.refer is overly simplistic and should look for a NOTIFY that contains the result of the attempted REFER. On a 2XX response to a REFER with replaces, the Session immediately terminates. 

```
    this.sendRequest(SIP.C.REFER, {
      extraHeaders: extraHeaders,
      body: options.body,
      receiveResponse: function (response) {
        if ( ! /^2[0-9]{2}$/.test(response.status_code) ) {
          return;
        }
        // hang up only if we transferred to a SIP address
        if (withReplaces || (target.scheme && target.scheme.match("^sips?$"))) {
          this.terminate();
        }
      }.bind(this)
    });
```

The Session needs the ability to continue to stay up to respond and act on the actual NOTIFY. The original 202 response simply means it's subscribed.

**Comment by klarrimore** [comment]
This obviously has lots of issues but I've been experimenting with waiting for the refer like this.

```
  this.sendRequest(SIP.C.REFER, {
      extraHeaders: extraHeaders,
      body: options.body,
      receiveResponse: function (response) {
        if ( ! /^2[0-9]{2}$/.test(response.status_code) ) {
          return;
        }
        // hang up only if we transferred to a SIP address
        if (withReplaces || (target.scheme && target.scheme.match("^sips?$"))) {
          if (response.status_code === 202) {
            this.once('notify', function(request) {
              if (request.call_id ===  response.call_id &&
                  request.to_tag === response.from_tag &&
                  request.from_tag ===  response.to_tag &&
                  request.hasHeader('Event') &&
                  /refer/.test(request.getHeader('Event'))) {
                    if (callback && /200/.test(request.body)) {
                      callback(true);
                    } else if (callback) {
                      callback(false);
                    }
              }
            })
          } else {
            this.terminate();
          }
        }
      }.bind(this)
```

**Comment by fredguth** [comment]
@klarrimore any news on this one?

**Comment by james-criscuolo** [comment]
We've made [a few recent changes to the refer logic](https://github.com/onsip/SIP.js/commit/fa6e596ee9998780f01ee5eaf55e7b75a9b16860) that will make the handling slightly better, but still will not wait for the subscription. We've seen most/all of the hardphones we interact with behave similarly to how SIP.js treats REFERs, so we decided not waiting for REFER subscriptions was "acceptable," but I'm sure there are some cases where this can be problematic. In those cases, I'd be curious how many phones actually work correctly.
